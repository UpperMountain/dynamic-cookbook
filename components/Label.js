import React from "react";
import { Text } from "react-native";
import Padded from "./Padded";
import theme from "../lib/theme";

export const Label = ({ children, light }) => {
  const color = light ? "white" : "black";
  return (
    <Padded bottom={2 / 3} top={4 / 3}>
      <Text style={[theme.label, { color }]}>{children.toUpperCase()}</Text>
    </Padded>
  );
};

export const Section = ({ children, light }) => (
  <Padded horizontal>
    <Label light={light}>{children}</Label>
  </Padded>
);
