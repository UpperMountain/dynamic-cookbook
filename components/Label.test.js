import React from "react";
import renderer from "react-test-renderer";
import { Label, Section } from "./Label";

describe("Label", () => {
  it("should render something without exploding", () => {
    renderer.create(<Label>Test Label</Label>);
  });

  it("should render a BLACK label", () => {
    const cmp = renderer.create(<Label>Default Test</Label>);
    expect(cmp.toJSON()).toMatchSnapshot();
  });

  it("should render a WHITE label", () => {
    const cmp = renderer.create(<Label light>Light Test</Label>);
    expect(cmp.toJSON()).toMatchSnapshot();
  });
});

describe("Section", () => {
  it("should render something without exploding", () => {
    renderer.creaate(<Section>Test Section</Section>);
  });

  it("should render a section label", () => {
    const cmp = renderer.create(<Section>Test</Section>);
    expect(cmp.toJSON()).toMatchSnapshot();
  });
});
