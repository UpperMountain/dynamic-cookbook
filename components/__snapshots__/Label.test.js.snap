// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render a BLACK label 1`] = `
<View
  style={
    Object {
      "paddingBottom": 13.333333333333332,
      "paddingLeft": NaN,
      "paddingRight": NaN,
      "paddingTop": 26.666666666666664,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(255,255,255,0.9)",
          "fontFamily": "raleway",
          "fontSize": 16,
          "letterSpacing": 1,
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    DEFAULT TEST
  </Text>
</View>
`;

exports[`Label should render a WHITE label 1`] = `
<View
  style={
    Object {
      "paddingBottom": 13.333333333333332,
      "paddingLeft": NaN,
      "paddingRight": NaN,
      "paddingTop": 26.666666666666664,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(255,255,255,0.9)",
          "fontFamily": "raleway",
          "fontSize": 16,
          "letterSpacing": 1,
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    LIGHT TEST
  </Text>
</View>
`;

exports[`Section should render a section label 1`] = `
<View
  style={
    Object {
      "paddingBottom": NaN,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": NaN,
    }
  }
>
  <View
    style={
      Object {
        "paddingBottom": 13.333333333333332,
        "paddingLeft": NaN,
        "paddingRight": NaN,
        "paddingTop": 26.666666666666664,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(255,255,255,0.9)",
            "fontFamily": "raleway",
            "fontSize": 16,
            "letterSpacing": 1,
          },
          Object {
            "color": "black",
          },
        ]
      }
    >
      TEST
    </Text>
  </View>
</View>
`;
